### ISSUE TITLE
Timeline based animation editor for Cut Scenes, CGI, etc.

### ISSUE THREAD

**Issue by HeadClot** [godotengine/godot#7765]
Hey Godot Devs,

With the upcoming 3.0 update we are getting an overhauled renderer that supports PBR and better Global Illumination. Would it be possible post 3.0 to get a Timeline based animation editor similar to [After Effects](http://www.indietips.com/wp-content/uploads/2013/02/After-Effects-Iron-Man-Avengers.jpg), [UE4 Sequencer](http://i.imgur.com/CmGhnLv.jpg), or [Timeline](https://blogs.unity3d.com/wp-content/uploads/2016/11/10-Timeline.jpg)?

I am requesting this as I have a need to make cut scenes in my projects.

**Comment by Zylann** [comment]
That can be already done already with AnimationPlayer, which is able to animate any property and even call functions over time (including starting other animations). The interface is a bit more cumbersome to work with curves though, as mentionned in other places (I don't remember).
Your request is a bit vague because you don't explain in detail what you want to add.

**Comment by HeadClot** [comment]
Hey Zylann,
From my understanding the Animation Player is just an key frame based animation node. That is what the docs are telling me at least.

I do not just need key framing but the ability to call individual imported animations (such as a Walk cycle, a attack animation, etc.) and blend them between other animations in editor. Or just cut a part of the animation out. For example If I want to cut the last 5 seconds of a 25 second animation out. I should be able to do that.

I would highly recommend looking at this video to see what can be improved in the AnimationPlayer Node or possibly via another tool in Godot. I would recommend watching the video linked to get a better idea of what I am trying to say. -> https://www.youtube.com/watch?v=5KSaHcsSVa4

Basically I am looking for a more advanced cut scene editor. Similar of that to Unity's timeline or Unreal Engine's Sequencer.

While Keyframe animation is nice. I do not want to have to redo existing animations just to get a shot right. This [wikipedia article](https://en.wikipedia.org/wiki/Non-linear_editing_system) should go into more detail on what I would like to see in godot. 

Sorry for being vague initially.
HeadClot

**Comment by bojidar-bg** [comment]
@HeadClot Maybe you want Animation*Tree*Player then, which would blend between all kinds of animations?

**Comment by TeddyDD** [comment]
I think I get it. You want nested animations - something like sequences in After Effects. You can do this with current animation system using function track (call function that starts animation of object) or you can animate properties of AnimationPlayer. Support for this kind of things in animation editor could be nice but I guess its a lot of work

**Comment by HeadClot** [comment]
Ok so - I think I found a way to sum this up nicely  a "director mode" built into godot.

**Comment by groud** [comment]
I think it is called "a sequencer"

**Comment by akien-mga** [comment]
`AnimationTreePlayer` is deprecated in Godot 3.1 and will be removed in Godot 3.2. It is replaced by `AnimationTree`, which was written from scratch and likely fixes/deprecates this issue.

If the issue is still relevant in some form with `AnimationTree` in Godot 3.1, please open a new issue about it (potentially linking this one if there is relevant discussion to keep track of).
